keras/metrics.py

```diff
@@ -3490,6 +3492,10 @@ def categorical_accuracy(y_true, y_pred):
   Returns:
     Categorical accuracy values.
   """
+  # assert if predicted and true labels Tensors have the same shape
+  tf.debugging.assert_equal(tf.shape(y_pred),
```

Comment:
Please remove this change

Reply:
This is from [#48000](https://github.com/tensorflow/tensorflow/pull/48000). Resolves [#46953](https://github.com/tensorflow/tensorflow/issues/46953) in TF repo.

Reply:
It is done incorrectly. We should not use `assert_equal` for user-facing error messages, and you're importing private APIs. Revert this file.

Reply:
Understood. I'll revert the change.